src/svd/util.py

```diff
@@ -0,0 +1,386 @@
+                config[peripheral.name] = {}
+
+        return config
+
+    def apply_memory(self, content: dict[int, int]) -> DeviceBuilder:
+        """Update device content based on a memory map.
+
+        The content is assumed to be at byte granularity and sorted by ascending address.
+
+        :param content: content memory map.
+        :returns: builder instance.
+        """
+        periph_map = self._device_periph_map()
+        reg_map = {}
+        current_periph = None
+        current_periph_range = range(-1, 0)
+        current_periph_regs = {}
+
+        map_iter = iter(content.items())
+
+        while True:
+            try:
+                addr_0, val_0 = next(map_iter)
+            except StopIteration:
+                break
+
+            if addr_0 not in current_periph_range:
+                for periph_range, periph in periph_map:
+                    if addr_0 in periph_range:
+                        current_periph = periph
+                        current_periph_range = periph_range
+                        periph_id = _get_periph_id(periph)
+                        if periph_id in reg_map:
+                            current_periph_regs = reg_map[periph_id]
+                        else:
+                            current_periph_regs = self._periph_reg_map(current_periph)
+                            reg_map[periph_id] = current_periph_regs
+                        self._written_peripherals.setdefault(periph_id, periph)
+                        break
+                else:
+                    svd.log.warning(
+                        f"Address 0x{addr_0:08x} does not correspond to any peripheral"
+                    )
+                    continue
+
+            assert current_periph_regs is not None
+
+            try:
+                reg = current_periph_regs[addr_0]
+            except KeyError:
+                svd.log.warning(
+                    f"Address 0x{addr_0:08x} is within the address range of {current_periph} "
+                    f"[0x{current_periph_range.start:x}-0x{current_periph_range.stop:x}), but "
+                    "does not correspond to any register in the peripheral"
+                )
+                continue
+
+            reg_len = reg.bit_width // 8
+            reg_content_bytes = [val_0]
+            for i in range(reg_len - 1):
+                try:
+                    _, val_i = next(map_iter)
+                    reg_content_bytes.append(val_i)
+                except StopIteration:
+                    raise ValueError(
+                        f"Content for {reg} was only partially specified. "
+                        f"Missing value for address 0x{addr_0 + i:08x}"
+                    )
+
+            # TODO: don't need to call this more than once
+            if not self._enforce_svd_constraints:
+                reg.unconstrain()
+
+            reg.content = int.from_bytes(bytes(reg_content_bytes), byteorder="little")
```

Comment:
Non-blocking: should `byteorder` depend on `Cpu.endian`?
(I'm not sure how svada handles endianness currently; if it's up to numpy, then it could be native endian...)